edith-step/src/ReceiptInput.js

```diff
@@ -9,15 +9,54 @@ export default class ReceiptInput extends React.Component {
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
-  handleSubmit(e) {
+  async getDeal(name, price, quantity) {
+    const axios = require('axios')
+    const response = await axios({
+      method: 'post',
+      url: '/receipt-deals',
+      data: {
+        itemName: name,
+        itemPrice: price,
+        itemQuantity: quantity
+      }
+    });
+    const dealItem = response.data;
+
+    let newDeal;
+    if (dealItem == "no deal found") {
+      newDeal = {
+        storeName: "no deal found",
+        storePrice: 0
+      }
+    } else {
+      newDeal = {
+        storeName: dealItem.store,
+        storePrice: dealItem.price
+      };
+    }
```

Comment:
Nit: It's better to use ```===``` instead of ```==``` since ```==``` does not check for data types. Also you might be able to use ternary operator to reduce the amount of lines here.
```
dealItem === "no deal found" ? newDeal = {storeName: "no deal found", storePrice: 0}
                             : newDeal = {storeName: dealItem.store, storePrice: dealItem.price};
```

Reply:
```javascript
const newDeal = dealItem === "no deal found" ? {storeName: "no deal found", storePrice: 0} : {storeName: dealItem.store, storePrice: dealItem.price};
```